stl/inc/flat_set

```diff
@@ -294,9 +294,10 @@ public:
         return _Erase(_Val);
     }
     template <class _Other>
-        requires _Keylt_transparent
+        requires (
+            _Keylt_transparent && !is_convertible_v<_Other &&, iterator> && !is_convertible_v<_Other &&, iterator>)
     size_type erase(_Other&& _Val) {
-        return _Erase(_STD forward<_Other>(_Val));
+        return _Erase(_Val);

```

Comment:
I believe it's safe to use `const auto&` in `_Erase`, as the `auto&&` in `erase` seems only to serve for constraint dispatching.